central/reports/common/query_builder.go

```diff
@@ -1,47 +1,64 @@
 
 import (
+	"context"
 	"fmt"
 	"strings"
 	"time"
 
+	collectionDataStore "github.com/stackrox/rox/central/resourcecollection/datastore"
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
 	"github.com/stackrox/rox/pkg/sac/effectiveaccessscope"
 	"github.com/stackrox/rox/pkg/search"
 )
 
 // ReportQuery encapsulates the cve specific fields query, and the resource scope
 // queries to be used in a report generation run
 type ReportQuery struct {
-	CveFieldsQuery string
-	ScopeQueries   []string
+	CveFieldsQuery   string
+	ScopeQueries     []string
+	DeploymentsQuery *v1.Query
 }
 
 type queryBuilder struct {
-	clusters              []*storage.Cluster
-	namespaces            []*storage.NamespaceMetadata
-	scope                 *storage.SimpleAccessScope
-	vulnFilters           *storage.VulnerabilityReportFilters
-	lastSuccessfulRunTime time.Time
+	clusters                []*storage.Cluster
+	namespaces              []*storage.NamespaceMetadata
+	scope                   *storage.SimpleAccessScope
+	vulnFilters             *storage.VulnerabilityReportFilters
+	collection              *storage.ResourceCollection
+	collectionQueryResolver collectionDataStore.QueryResolver
+	lastSuccessfulRunTime   time.Time
 }
 
 // NewVulnReportQueryBuilder builds a query builder to build scope and cve filtering queries for vuln reporting
 func NewVulnReportQueryBuilder(clusters []*storage.Cluster,
-	namespaces []*storage.NamespaceMetadata, scope *storage.SimpleAccessScope,
-	vulnFilters *storage.VulnerabilityReportFilters, lastSuccessfulRunTime time.Time) *queryBuilder {
+	namespaces []*storage.NamespaceMetadata, scope *storage.SimpleAccessScope, collection *storage.ResourceCollection,
+	vulnFilters *storage.VulnerabilityReportFilters, collectionQueryRes collectionDataStore.QueryResolver,
+	lastSuccessfulRunTime time.Time) *queryBuilder {
 	return &queryBuilder{
-		clusters:              clusters,
-		namespaces:            namespaces,
-		scope:                 scope,
-		vulnFilters:           vulnFilters,
-		lastSuccessfulRunTime: lastSuccessfulRunTime,
+		clusters:                clusters,
+		namespaces:              namespaces,
+		scope:                   scope,
+		vulnFilters:             vulnFilters,
+		collection:              collection,
+		collectionQueryResolver: collectionQueryRes,
+		lastSuccessfulRunTime:   lastSuccessfulRunTime,
 	}
 }
 
 // BuildQuery builds scope and cve filtering queries for vuln reporting
-func (q *queryBuilder) BuildQuery() (*ReportQuery, error) {
-	scopeQueries, err := q.buildScopeQueries()
+func (q *queryBuilder) BuildQuery(ctx context.Context) (*ReportQuery, error) {
+	var err error
+	var deploymentsQuery *v1.Query
+	var scopeQueries []string
+	if !features.ObjectCollections.Enabled() {
+		scopeQueries, err = q.buildScopeQueries()
+	} else {
+		deploymentsQuery, err = q.collectionQueryResolver.ResolveCollectionQuery(ctx, q.collection)
+	}
```

Comment:
Flip it to make it easier to parse.
```suggestion
	if features.ObjectCollections.Enabled() {
		deploymentsQuery, err = q.collectionQueryResolver.ResolveCollectionQuery(ctx, q.collection)
	} else {
		scopeQueries, err = q.buildScopeQueries()
	}
```

Reply:
+1